File: clojure/raindrops/src/raindrops.clj
(ns raindrops)

(defn word [number literal divisor]
  (if (= 0 (rem number divisor))
    literal
    ""))

(defn plingplangplong [number]
  (str (word number "Pling" 3) (word number "Plang" 5) (word number "Plong" 7)))

(defn convert
  [number]   
  (let [res (plingplangplong number)]
    (if (= res "")
      (str number)
      res)))

File: go/raindrops/raindrops.go
package raindrops

import (
  "strconv"
  "fmt"
)

const testVersion = 3

// Convert converts number to string using raindrop rules
func Convert(x int) string {
  str := fmt.Sprintf("%s%s%s", pling(x), plang(x), plong(x))
  if len(str) == 0 {
    return strconv.Itoa(x)
  }
  return str
}

func pling(x int) string {
  if x % 3 == 0 {
    return "Pling"
  }
  return ""
}

func plang(x int) string {
  if x % 5 == 0 {
    return "Plang"
  }
  return ""
}

func plong(x int) string {
  if x % 7 == 0 {
    return "Plong"
  }
  return ""
}
